airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/LauncherWorker.java

```diff
@@ -117,12 +119,30 @@ public OUTPUT run(INPUT input, Path jobRoot) throws WorkerException {
             containerOrchestratorConfig.containerOrchestratorImage(),
             containerOrchestratorConfig.googleApplicationCredentials());
 
+        cancellationCallback.set(() -> {
+          // When cancelled, try to set to true.
+          // Only proceed if value was previously false, so we only have one cancellation going. at a time
+          if (!isCanceled.getAndSet(true)) {
+            log.info("Trying to cancel async pod process.");
+            process.destroy();
+          }
+        });
+
+        // only kill running pods and create process for the first run for an attempt
         if (process.getDocStoreStatus().equals(AsyncKubePodStatus.NOT_STARTED)) {
```

Comment:
version of this code when would this ever _not_ be equal to `NOT_STARTED`. didn't we always kill it?

Reply:
Unless the attempt was interrupted while spinning up the pod, any re-run of the launcher for the same attempt should skip the creation process since it already exist.